server/user.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * @fileoverview user function handler
+ */
+var fs = require('fs');
+var assert = require('assert');
+
+var log = require('./log.js');
+var utils = require('./utils.js');
+
+/** @type {user} */
+module.exports = user;
+
+/**
+ * @constructor
+ */
+function user() {}
+
+/** @enum {string} */
+user.QueryType = {
+  SIGNUP: 'sign-up',
+  SIGNIN: 'sign-in'
+};
+
+/**
+ * Expire time of cookie.
+ * @const {number}
+ */
+user.cookieExpireTime = 24 * 60 * 60 * 1000;
+
+/**
+ * @typedef {{
+ *   email: (string|undefined),
+ *   username: string,
+ *   password: string,
+ *   confirmed: (boolean|undefined)
+ * }}
+ */
+user.Info;
+
+/**
+ * @typedef {{
+ *   error: string
+ * }}
+ */
+user.Error;
+
+/**
+ * Checks the user information for signing in.
+ * @param {!mongodb.Db} db Session database.
+ * @param {string} userPath Directory of user information file.
+ * @param {!user.Info} userInfo User Information
+ * @param {function(!Object)} callback Callback function.
+ */
+user.signUp = function(db, userPath, userInfo, callback) {
```

Comment:
As I said before, I think only client side validation is necessary. Because if the username/password is not valid, we should tell the user at the client side when he inputs registration information and revises it. Not after the user has submitted the input already and you tell him that it is not valid and please retype everything again.